src/graph/utils/sycl_check.hpp

```diff
@@ -26,18 +26,8 @@
 #error "Unsupported compiler"
 #endif
 
-#if defined(__INTEL_LLVM_COMPILER)
-#if (__INTEL_LLVM_COMPILER < 20230000)
-#define DNNL_USE_SYCL121_API 1
-#else
-#define DNNL_USE_SYCL121_API 0
-#endif
-#elif defined(__LIBSYCL_MAJOR_VERSION)
-#if (__LIBSYCL_MAJOR_VERSION < 6)
-#define DNNL_USE_SYCL121_API 1
-#else
-#define DNNL_USE_SYCL121_API 0
-#endif
+#if defined(__INTEL_LLVM_COMPILER) && __INTEL_LLVM_COMPILER >= 20230000
+#elif defined(__LIBSYCL_MAJOR_VERSION) && __LIBSYCL_MAJOR_VERSION >= 6
```

Comment:
(random) there might be other spots to edit:
- check SYCL_COMPILER_VERSION macro
- there might be some spots in cuda/hip backend as well where workarounds were implemented (tagging @t4c1 @sgeor255).